src/mail/view/MailView.ts

```diff
@@ -652,27 +640,73 @@ export class MailView implements CurrentView {
 		moveMails({mailModel: locator.mailModel, mails: mailsToMove, targetMailFolder: folder})
 	}
 
-	private createFolderAddButton(mailGroupId: Id, parentFolder: MailFolder | null, folderSystem: FolderSystem): IconButtonAttrs {
+	private createFolderAddButton(mailGroupId: Id, parentFolder: MailFolder | null): IconButtonAttrs {
 		return {
 			title: "addFolder_action",
 			click: () => {
-				return this.showCreateFolderDialog(mailGroupId, parentFolder, folderSystem)
+				return this.showFolderAddEditDialog(mailGroupId, null, parentFolder)
 			},
 			icon: Icons.Add,
 			size: ButtonSize.Compact,
 		}
 	}
 
-	private showCreateFolderDialog(mailGroupId: string, parentFolder: MailFolder | null, folderSystem: FolderSystem) {
-		const infoMsg = parentFolder ? () => getPathToFolderString(folderSystem, parentFolder) : null
-		const parentFolderId: IdTuple | null = parentFolder ? parentFolder._id : null
-		Dialog.showProcessTextInputDialog("folderNameCreate_label", "folderName_label", infoMsg, "", async (name) => {
-				if (parentFolderId) {
-					this.expandedState.add(elementIdPart(parentFolderId))
+	private async showFolderAddEditDialog(mailGroupId: string, folder: MailFolder | null = null, parentFolder: MailFolder | null = null) {
+		const noParentFolderOption = "No Parent Folder";
+		const folderNameValue = stream(folder?.name ? folder?.name : "")
+		// TODO getMailboxDetails wont work for shared mailboxes?
```

Comment:
yea this will only return the mailbox where currently a folder is selected or the users own mailbox.
I guess you could call `locator.mailModel.getMailboxDetailsForMailGroup` with the passed in `mailGroupId`